src/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_resolver.py

```diff
@@ -212,24 +235,51 @@ def resolve(self, dict, key):
         raise UnableToResolveVariableException()
 
     def key_to_str(self, key):
-        if isinstance(key, str):
-            return '%r' % key
-        else:
-            if not pydevd_constants.IS_PY3K:
-                if isinstance(key, unicode):
-                    return "u'%s'" % key
-            return key
+        if not pydevd_constants.IS_PY3K:
+            if isinstance(key, unicode):
+                return "u'%s'" % key
```

Comment:
This looks like it'd have issues with quotes, backslashes, and other stuff that might need escaping.

Reply:
Also, what if the key is a tuple - won't that just take the first item's repr? I think this needs to be `(key,)` to ensure it works correctly in all cases, and same for the line below.

Reply:
Agreed (fixed -- always using `'%r' % (key,)`).